### PR TITLE
Allow line style and shape on radar charts to be defined

### PR DESCRIPTION
**Pull request by SelcalHF**
The intent is to allow overriding the default autocycle of line color and shape type on radar charts. As it stands, charts with the same dataset will not have the same symbols as the radar cycles them. In addition, the request modifies the legend drawing to allow the same, otherwise these will not match.
I've tried to keep to the style of the original code, however, fully realize this may not be a good fit. But since I'm a happy user of your recode, I at least want to highlight my modifications and intent in case it can help.

### CODE REVIEW

**Comment by szymach** [comment]
Hello!
Thank you for your submission. Could you perhaps post images of charts generated before and after your changes? The library still lacks tools to visualize what the code changes actually do. Thank you.

**Comment by SelcalHF** [comment]
--Hi,

Of course!

The intend is to allow, using series properties, define the shape and line style of radar charts.
This is because color-blind people otherwise have no way of reading the chart.

Original 3.0 code produces a fixed shape (filled circle) in the graphs for all series:
(ignore the lack of shapes in the legend, that is due to the call setting a legend family that is not implemented in the original code)
afbeelding

afbeelding


As you can see, all shapes are circles and the lines are quite thin. Mainly for color-blind users, the round shapes aren't effective as now interpretation depends only on line color and they cannot see it.

So, the code change allows to set the shape (and line style):
afbeelding

afbeelding


The defined shapes show in both the legend and the graph. Also, the lines now follow the defined line-style.

No changes would be observed by users that don't set any of the options. In other words, only if the specific values are set, does the output change.
I've tried to make this equal to setting these options for other graph types.

In the code, to set the shape, the usual call is used:       

`$myData->setSerieShape($serieID,  SERIE_SHAPE_FILLEDTRIANGLE );`

And to draw the legend, pass the new family constant to let the drawing function retrieve the associated series shapes:
`
$myPicture->drawLegend($legendX, $legendY, [ "Style"=>LEGEND_BOX,"Mode"=>LEGEND_VERTICAL,"Family"=>LEGEND_FAMILY_ASDEFINED ]);`

Hope this makes it clear what I've been trying to do :)

Thanks for your hard work, your branch has really, REALLY saved me so much work!

**Comment by szymach** [comment]
Hello!
I have created a new branch, `3.1` and have changed the base of your request to point at it, since this is a new feature. Could you rebase your PR onto it? Thank you.

**Comment by SelcalHF** [comment]
Done -- thanks!

**Comment by szymach** [comment]
I have restored Travis builds (there was an error on their part and I had to set it straight) and it seems you have some CS fixes and [an error](https://app.travis-ci.com/github/szymach/c-pchart/jobs/604340740#L326) to fix :) Remember this has to also run on PHP 5.4, lots of folk run it on outdated setups.

File: src/Chart/Radar.php [hunk 1]
```diff
@@ -535,7 +536,7 @@ public function drawRadar(Image $Object, Data $Values, array $Format = [])
                     $Points[count($Points) - 1][1],
                     $Points[0][0],
                     $Points[0][1],
-                    $Color
+                    $Color + (isset($Format['LineWeight']) ? [ 'Weight' => $Format['LineWeight'] ] : [] ),
```
Comment (szymach) [inline src/Chart/Radar.php]: Please remove the trailing `,`, it is not supported on older PHP versions.

File: src/Chart/Radar.php [hunk 2]
```diff
@@ -547,16 +548,26 @@ public function drawRadar(Image $Object, Data $Values, array $Format = [])
...
+                        1,
+                        $Color['R'],
+                        $Color['G'],
+                        $Color['B'],
+                        $Color['Alpha'],
+                        0,
+                        0,
+                        0,
+                        100
+                    );
```
Comment (szymach) [inline src/Chart/Radar.php]: There seems to be a whitespace on this line.

Comment (SelcalHF) [reply src/Chart/Radar.php]: VScode keeps adding some here and there, I think I've found the extention that does it... Thanks!

File: src/Chart/Radar.php (hunk 1 above)
Comment (SelcalHF) [reply src/Chart/Radar.php]: Done

**Comment by SelcalHF** [comment]
Sorry I'm making a mess because I'm trying to hurry. I'll fix it properly after lunch :)

**Comment by szymach** [comment]
No worries, take your time. You can easily run these scripts locally, too :)

**Comment by SelcalHF** [comment]
Actually turns out the last commit did fix the last issues :).
You know that feeling, that you think "I'll quickly do this, because I know it'll be fine" while it never is? 😄 
Sorry about that!

**Comment by szymach** [comment]
Well actually I was checking if I have to use paid credits to run builds on public repositories and it turns out that I do :) So your last push simply did not register. Can you amend your commit and force push it to trigger the build?

**Comment by SelcalHF** [comment]
Done & passed!

**Review by szymach — APPROVED** [review]

**Comment by szymach** [comment]
Thank you, I will pack a release sometime soon :)

**Pull request merged**